Tests/HashTreeCollectionsTests/TreeDictionary.Keys Tests.swift

```diff
@@ -105,5 +105,66 @@ class TreeDictionaryKeysTests: CollectionTestCase {
       }
     }
   }
+  
+  func test_isEqual_exhaustive() {
+    withEverySubset("a", of: testItems) { a in
+      let x = TreeDictionary<RawCollider, Int>(
+        uniqueKeysWithValues: a.lazy.map { ($0, 2 * $0.identity) })
+      let u = Set(a)
+      expectEqualSets(x.keys, u)
+      withEverySubset("b", of: testItems) { b in
+        let y = TreeDictionary<RawCollider, Int>(
+          uniqueKeysWithValues: b.lazy.map { ($0, -$0.identity - 1) })
+        let v = Set(b)
+        expectEqualSets(y.keys, v)
+
+        let reference = u == v
+        print(reference)
+
+        expectEqual(x.keys == y.keys, reference)
+      }
+    }
+  }
+  
+  func test_Hashable() {
+    let strings: [[[String]]] = [
+      [
+        []
+      ],
+      [
+        ["a"]
+      ],
+      [
+        ["b"]
+      ],
+      [
+        ["c"]
+      ],
+      [
+        ["d"]
+      ],
+      [
+        ["e"]
+      ],
+      [
+        ["a", "b"], ["b", "a"],
+      ],
+      [
+        ["a", "d"], ["d", "a"],
+      ],
+      [
+        ["a", "b", "c"], ["a", "c", "b"],
+        ["b", "a", "c"], ["b", "c", "a"],
+        ["c", "a", "b"], ["c", "b", "a"],
+      ],
+      [
+        ["a", "d", "e"], ["a", "e", "d"],
+        ["d", "a", "e"], ["d", "e", "a"],
+        ["e", "a", "d"], ["e", "d", "a"],
+      ],
+    ]
+    let keys = strings.map { $0.map { TreeDictionary(uniqueKeysWithValues: $0.map { ($0, UUID().uuidString) }).keys }}
+    checkHashable(equivalenceClasses: keys)
```

Comment:
@lorentey How about this? I'm generating random values with `UUID().uuidString` so we can confirm the `keys` are still equal when values are not equal. I could try something different if we prefer for those values to be generated deterministically (the same every time).

Reply:
Nice! I was actually typing in a comment recommending something along these lines -- a deterministic alternative would be to use a sequential counter, but randomizing things works just as well.

Instead of Foundation's UUID, I think we should just use `SystemRandomNumberGenerator`.